### PR TITLE
Generate mirror tarball

### PR DESCRIPTION
**Pull request by quaresmajose**
When building public LmP **major release** we will generate the mirror tarball on the download folder of all git repos.

All the download files can be updated to the public download mirror at the end:
```
rsync -av --no-r --exclude=*.done --exclude=*.tmp /downloads/ /download-mirror/
```

### CODE REVIEW

File: lmp/bb-build.sh [hunk 1]
```diff
@@ -67,6 +67,13 @@ status "Run bitbake (save the global and image specific environments)"
 bitbake -e > ${archive}/bitbake_global_env.txt
 bitbake -e ${IMAGE} > ${archive}/bitbake_image_env.txt
 
+# Public LmP build and we are building the major release
+if [[ "${H_PROJECT}" == "lmp" ]] && [[ "${LMP_VERSION_MINOR}" == "0" ]] ; then
+    status "Run bitbake (generate mirror tarball)"
+    echo 'BB_GENERATE_MIRROR_TARBALLS = "1"' >> conf/local.conf
+    bitbake --runall fetch --no-setscene ${IMAGE}
+fi
```
Comment (doanac) [inline lmp/bb-build.sh:75]: are we going to be saving this as a build artifact and if so, how expensive will this be?

Comment (quaresmajose) [reply lmp/bb-build.sh:75]: No, we will not save it as build artifacts for every build. The idea is to collect all the source archives need to build the public lmp when we do a release and upload them to the public https://storage.googleapis.com/lmp-cache/downloads/

The size of the _am62xx-evm_ machine is about 20G
```
du -sh downloads/
20G     downloads/
```

Comment (quaresmajose) [reply lmp/bb-build.sh:75]: The cleanup method can be a matter of remove the `/lmp-cache/downloads/` run a build for each active lmp release i.e: v92, v91, v90, v89, v88 etc and upload the worker downloads/ folder to the download mirror https://storage.googleapis.com/lmp-cache/downloads/

Comment (doanac) [reply lmp/bb-build.sh:75]: so its for you running builds locally? looks good to me

Comment (quaresmajose) [reply lmp/bb-build.sh:75]: Not really, this PR is meant to be run on CI in the public lmp factory in each major release.

For example the workflow of the next release will be:

1. We tag the v93 on the lmp-manifest
2. The next build and only the next will create an archive on the bibake download folder of each git repo source code used in the build
3. At the end of the build we should also rsync the files on the downloads/ folder additionally to sstate-cache we already sync. The last step needs to be added for the existing script that rsync the sstate-cache

After that we will no longer depend on external sources code and we will have everything we need in our public mirror https://storage.googleapis.com/lmp-cache/downloads/

**Review by doanac — APPROVED** [review]

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]

File: helpers.sh [hunk 2]
```diff
@@ -113,6 +113,7 @@ function set_base_lmp_version {
 
 	# 3: Find our base LMP version based on the HEAD
 	export LMP_VERSION=$(git describe --tags --abbrev=0 HEAD)
+	export LMP_VERSION_MINOR="$(git rev-list ${LMP_VERSION}..HEAD --count)"
```
Comment (angolini) [inline helpers.sh:116]: there is a typo on the commit lot "nunber"

**Review by angolini — APPROVED** [review]
I like it

File: helpers.sh (hunk 2 above)
Comment (quaresmajose) [reply helpers.sh:116]: fixed! thanks

**Review by igoropaniuk — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**